### PR TITLE
OpenTracing - Error handling when span starting fails so requests continue

### PR DESCRIPTION
**Pull request by tonyreigns**
Errors when a span is started are now caught so the request continues and the trace error is outputted rather than the entire request failing due to a failed trace.

See issue: #17875

### CODE REVIEW

**Comment by wraschke** [comment]
#run-libby

**Comment by tonyreigns** [comment]
#build

**Review by fmhwong — CHANGES_REQUESTED** [review]
Can you add a FAT?

**Comment by fmhwong** [comment]
@tonyreigns As discussed, please limit the number of error messages printed to the log.

**Comment by tonyreigns** [comment]
#build

File: dev/com.ibm.ws.opentracing.1.1/src/com/ibm/ws/opentracing/OpentracingClientFilter.java [hunk 1]
```diff
@@ -129,19 +131,26 @@ public void filter(ClientRequestContext clientRequestContext) throws IOException
...
-
-            try (AutoFinishScope scope = AUTO_FINISH_SCOPE_MANAGER.activate(span, true)) {
-                Continuation continuation = scope.capture();
-
-                tracer.inject(
-                              scope.span().context(),
-                              Format.Builtin.HTTP_HEADERS, new MultivaluedMapToTextMap(clientRequestContext.getHeaders()));
-
-                clientRequestContext.setProperty(CLIENT_CONTINUATION_PROP_ID, continuation);
+            try{
```
Comment (fmhwong) [inline dev/com.ibm.ws.opentracing.1.1/src/com/ibm/ws/opentracing/OpentracingClientFilter.java:134]: Can you fix the indent with 4 space?   Please don't use tab.

File: dev/com.ibm.ws.opentracing.1.1/src/com/ibm/ws/opentracing/OpentracingClientFilter.java [hunk 2]
```diff
@@ -129,19 +131,26 @@ public void filter(ClientRequestContext clientRequestContext) throws IOException
...
+                  Continuation continuation = scope.capture();
+
+                  tracer.inject(
+                                scope.span().context(),
+                                Format.Builtin.HTTP_HEADERS, new MultivaluedMapToTextMap(clientRequestContext.getHeaders()));
+
+                  clientRequestContext.setProperty(CLIENT_CONTINUATION_PROP_ID, continuation);
+              }
+            } catch (Error e){
+                if(!spanErrorLogged) {
```
Comment (fmhwong) [inline dev/com.ibm.ws.opentracing.1.1/src/com/ibm/ws/opentracing/OpentracingClientFilter.java]: We usually have a space after `if`

**Review by fmhwong — CHANGES_REQUESTED** [review]
Is `OPENTRACING_COULD_NOT_START_SPAN` a new message?  I don't see it added to `Opentracing.nlsprops`.  Also what is the Exception you are going to catch?  Can we narrow it down a bit as you're just catching all `Error`s.  I worry that once an Error is thrown, we will hide all other `Error` and `Exception`.

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#build

**Review by fmhwong — APPROVED** [review]
LGTM

**Comment by fmhwong** [comment]
@tonyreigns You might need a message review.

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#libby

**Comment by skasund** [comment]
L2 message review completed

**Comment by tonyreigns** [comment]
#build

**Comment by tonyreigns** [comment]
#libby

**Pull request merged**